### DISCUSSION TITLE
Install procedure

### DISCUSSION THREAD

**Discussion by ghost** [ltsp/ltsp#321] [Q&A] [answered]
So I followed the install procedure and everything went fine, but my client machine (a x86 PC with one NIC) will not boot. 
It wont find the server. I have one NIC on the server and a gigabit switch. 
Do i need to make any other network configuration after the install or it should just work ? How can I check to see if all LTSP services are working  ? 

Thanks.

**Comment by alkisg** [comment]
The most common problem is that people skip one of the installation steps. Did you run `ltsp image /`, `ltsp dnsmasq` and all the other commands?

What is the output of the following commands on the server?

```
sudo -i
ltsp info
ip a
/usr/lib/klibc/bin/ipconfig -n enp2s0
```

In the last one, replace enp2s0 with the correct interface name.

Finally, what message do the clients show when they fail to contact the server?

**Reply by ghost** [reply]
Hi, yes executed all the commands.
 
I'm using kubuntu 20.04 on the server. I tried two different installs one with ltsp dnsmasq and the last one with ltsp dnsmasq --proxy-dhcp=0 (the one bellow). The client (tried with two different motherboards) tries to find a connection and doesn't give me any feedback.

[sudo] password for pharaoh: 
root@pharaoh-server:~# ltsp info
LTSP version: 20.06-1~ubuntu20.04.1
SERVER XSESSIONS: plasma.desktop
SERVER OS: PRETTY_NAME="Ubuntu 20.04 LTS"
CHROOTS:
VMs:
IMAGES:
x86_64
root@pharaoh-server:~# ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: enp2s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether 7e:65:89:6c:90:a1 brd ff:ff:ff:ff:ff:ff
    inet 192.168.67.1/24 brd 192.168.67.255 scope global noprefixroute enp2s0
       valid_lft forever preferred_lft forever
    inet6 fe80::7c65:89ff:fe6c:90a1/64 scope link 
       valid_lft forever preferred_lft forever
root@pharaoh-server:~# /usr/lib/klibc/bin/ipconfig -n enp2s0
IP-Config: enp2s0 hardware address 7e:65:89:6c:90:a1 mtu 1500 DHCP RARP

**Comment by alkisg** [comment]
The ipconfig command doesn't return an IP. Is your LTSP server supposed to be the DHCP server in this scenario? In this case, it's more usual to have a two-NIC setup... how does your LTSP server get an Internet connection, isn't there a router nearby?
Are you using virtual machines? If so, you'll need bridged mode...

**Reply by ghost** [reply]
Yes i want the server to also be the DHCP server, it have just one NIC, and no internet connection (is it obligatory ? I used an wireless dongle to install ltsp and removed it). They are physical machines.

**Comment by alkisg** [comment]
Sure, what you're asking for is fine and should work out of the box.

1. What's the output of `sudo netstat -nap | grep dnsmasq` on the ltsp server?
2. Can you upload a screenshot from the client boot failure?
3. If you boot the client with a live USB stick, does it get an IP from the ltsp server?

**Reply by ghost** [reply]
Here are the output for those commands.

1.:
root@pharaoh-server:~# sudo netstat -nap | grep dnsmasq
udp        0      0 0.0.0.0:4011            0.0.0.0:*                           629/dnsmasq         
udp        0      0 0.0.0.0:67              0.0.0.0:*                           629/dnsmasq         
udp        0      0 0.0.0.0:69              0.0.0.0:*                           629/dnsmasq         
udp6       0      0 :::69                   :::*                                629/dnsmasq         
unix  2      [ ]         DGRAM                    27125    629/dnsmasq  

3.:yes, according to ip a it gets the IP 192.168.67.132/24

2.:
[bios, boot, failure](https://postimg.cc/gallery/mcR9Vv1)

Thanks.

**Comment by alkisg** [comment, marked as answer]
I think that you enabled the LAN ROM but you didn't select "Boot from network first" in BIOS. See a screenshot of a failed PXE boot, do you see the client reporting that it got an IP?

https://filedb.experts-exchange.com/incoming/2014/08_w35/800_869192/PXE.JPG

Also can you run `journalctl -fu dnsmasq`, try to boot a client, and paste the journalctl output here?

**Comment by ghost** [comment]
Yes you are right, I changed the LAN options in the bios but not on the realtek setup (shift-f10). Now my client machine received an IP address like the one you sent.
I'm getting a new error now (error 0 TFTP Aborted received from 192.168.67.132) but i think this is similar to the discussion #276.  I will try to fix it here using the same solution and let you know. 

I just want to say that the dedication you put in this project is awesome, thanks a lot.

Update: Yep, it was the same problem as in discussion #276 my client is up and running now.
